File: 프로그래머스/lv0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.cpp
#include <string>
#include <vector>

using namespace std;

int array[31][31];

int solution(int balls, int share)
{
    int answer = 0;

    for (int i = 0; i < 31; i++)
        array[i][0] = array[i][i] = 1;

    for (int i = 2; i < 31; i++)
        for (int j = 1; j < i; j++)
            array[i][j] = array[i - 1][j] + array[i - 1][j - 1];

    answer = array[balls][share];

    return answer;
}

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.java
import java.math.BigInteger;

class Solution {

    public BigInteger fac(int i) {
        if (i == 1 || i == 0) {
            return BigInteger.ONE;
        }
        return BigInteger.valueOf(i).multiply(fac(i - 1));
    }

    public BigInteger solution(int balls, int share) {
        return fac(balls).divide(fac(share).multiply(fac(balls - share)));
    }
}